### PR TITLE
Add top level sections to the Unstable Book.

### PR DESCRIPTION
**Pull request by frewsxcv**
Prior to this commit, the contents of the Unstable Book were assumed to
be unstable features. This commit moves features into 'language features'
or 'library features' subsections. It also moves the 'linker_flavor'
compiler flag into a new 'Compiler Flags' subsection.

Even though it was helpful, I removed the tidy check that
cross-references the SUMMARY.md links with the Unstable Book directory
contents just because it would be difficult to maintain.

Relevant PR: https://github.com/rust-lang/rust/issues/41142.

### CODE REVIEW

File: src/tools/tidy/Cargo.toml [hunk 1]
```diff
@@ -2,6 +2,3 @@
 name = "tidy"
 version = "0.1.0"
 authors = ["Alex Crichton <alex@alexcrichton.com>"]
-
-[dependencies]
-regex = "0.2"
```
Comment (Mark-Simulacrum) [inline src/tools/tidy/Cargo.toml]: Looks like a Cargo.lock file here needs to be updated presumably because of this change.

**Comment by steveklabnik** [comment]
While we're doing this, should language features vs library features also happen?

**Comment by frewsxcv** [comment]
> While we're doing this, should language features vs library features also happen? 

Yeah, I can do that.

**Comment by frewsxcv** [comment]
@steveklabnik How does this look?

**Comment by frewsxcv** [comment]
If anyone has any idea what's going on with the linkchecker here, please let me know

**Comment by frewsxcv** [comment]
I fixed most of the tidy lint issues, I had to disable linkchecker for the Unstable Book (for now), since I'm using a subdirectory which mdbook uses `<base>` for which linkchecker doesn't support.

The remaining issues are here:

```
book/compiler-plugins.html:32: broken link - unstable-book/plugin.html
book/first-edition/print.html:8645: broken link - unstable-book/intrinsics.html
book/first-edition/print.html:13986: broken link - unstable-book/lang-items.html
book/first-edition/unsafe.html:188: broken link - unstable-book/intrinsics.html
book/first-edition/using-rust-without-the-standard-library.html:79: broken link - unstable-book/lang-items.html
reference/attributes.html:392: broken link - unstable-book/plugin.html
reference/macros.html:80: broken link - unstable-book/plugin.html
reference/print.html:559: broken link - unstable-book/plugin.html
reference/print.html:2193: broken link - unstable-book/plugin.html
```

These require changes upstream to the book and reference. What's the recommended approach? Set up redirects for everything?

**Comment by steveklabnik** [comment]
> These require changes upstream to the book and reference. What's the recommended approach? Set up redirects for everything?

We say this PR is good to go, then you update the submodules in this PR and ship it together

**Comment by frewsxcv** [comment]
> We say this PR is good to go

Sooooo is this PR good to go? ;) I can take care of the merge conflict, though they're going to keep popping up until this (and the book/reference PRs) gets shipped

**Comment by steveklabnik** [comment]
Yes, conceptually this PR is good to go; as long as you can take care of all of it within this week (due to beta branching), then let's do it.

**Comment by frewsxcv** [comment]
This is good to go now!

**Comment by steveklabnik** [comment]
@bors: r+ rollup

🎊

**Comment by frewsxcv** [comment]
@bors r=steveklabnik

fixed with https://github.com/rust-lang/rust/pull/41295/commits/2e2a670073b5abd523418357a2ebbbc1352ec696

**Pull request merged**